### PR TITLE
[DOP-4170]: Create AWS ECS Task definition to process Snooty Parser cache updates

### PR DESCRIPTION
**Pull request by branberry**
### Ticket
[DOP-4170](https://jira.mongodb.org/secure/RapidBoard.jspa?rapidView=953&projectKey=DOP&view=detail&selectedIssue=DOP-4170)

### Notes

This ticket implements the cache update worker via an ECS task. This ticket also contains additional updates to the `update-feature-branch.yml` workflow to process jobs in parallel, as well as provide caching for `node_modules` between jobs. 

[Spike document for context](https://docs.google.com/document/d/1ZJY8deWAqmAVrsvsb1emImerhqOT8q8EM4UJaPhCyWU/edit)
### Verification

Here is a [link](https://us-east-2.console.aws.amazon.com/cloudwatch/home?region=us-east-2#logsV2:log-groups/log-group/cache-updater-cacheupdaterresourcescacheUpdaterWorkerLogGroupD5F2B1BB-H8U91HUC6gWl/log-events/cacheupdater$252FcacheUpdaterWorkerImage$252Fc86e4ef0fa41417698dc041c49053f4c$3Fstart$3D1704736632944) to the CloudWatch logs that show a successful job run.

Here is a [link](https://s3.console.aws.amazon.com/s3/object/snooty-parse-cache?region=us-east-2&prefix=.snooty-java-0.15.2_cfb712a48071b0c14a2b099e61996398fd0b1382_76786132fde37e50ced10273a3062e43792a4d3b.cache.gz) of the uploaded cache file for `docs-java`.

### CODE REVIEW

File: cdk-infra/lib/constructs/api/webhook-api-construct.ts
Comment (mmeigs) [inline cdk-infra/lib/constructs/api/webhook-api-construct.ts:3]: Are these new imports needed still? I believe they're unused.

File: src/cache-updater/Dockerfile.cacheUpdater
Comment (mmeigs) [inline src/cache-updater/Dockerfile.cacheUpdater:47]: Just wondering why this is `docsworker` instead of `docsworker-xlarge`. I remember us having to change this for the local autobuilder to `docsworker-large` to keep consistent.

Comment (branberry) [reply src/cache-updater/Dockerfile.cacheUpdater:47]: I think I must've copied this from a previous version of the `Dockerfile.local`. It should be fine in this cause because it needs to be consistent within the Autobuilder because we use the username within the Autobuilder code. The code in this Dockerfile doesn't depend on the username at all, we just need a random user to place the AWS credentials.

File: cdk-infra/lib/constructs/cache-updater/cache-updater-worker-construct.ts
Comment (branberry) [reply cdk-infra/lib/constructs/cache-updater/cache-updater-worker-construct.ts]: Hey Matt! I think this is a great suggestion, but I am running into issues when importing it here. I believe it has to do with the fact that the `cdk-infra` is a separate project from the root project, and this leads to funky behavior. 

What I'll do is define the constant here, and add that as an environment variable 👍

**Review by seungpark — COMMENTED** [review]
this mostly LGTM. waiting for changes in last review by Matt and left a minor comment about requiring version / branch / tag detail of doc repo

File: src/cache-updater/index.ts
Comment (branberry) [reply src/cache-updater/index.ts:14]: Good point! From what I understand though, these cache files will only be for the master/main branch I believe... I'll double check though to confirm.

File: cdk-infra/lib/constructs/cache-updater/cache-updater-worker-construct.ts
Comment (mmeigs) [reply cdk-infra/lib/constructs/cache-updater/cache-updater-worker-construct.ts]: Ok! Def not a blocker!

File: src/cache-updater/index.ts
Comment (seungpark) [reply src/cache-updater/index.ts:14]: discussed offlinethis doc points out this will be for the master/main branch of the repo

**Review by seungpark — APPROVED** [review]
LGTM 👍

**Pull request merged**
